### PR TITLE
task/generate-pdf

### PR DESCRIPTION
**Pull request by kachmashk**
this endpoint creates catalog of passed products

### CODE REVIEW

File: api/helpers/pdfGenerate.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const folderPath = "./pdfs/"
+
+const localeDate = new Date().toLocaleDateString().split("/")
+const localeTime = new Date().toLocaleTimeString().split(":")
+
+const cutDate = `${localeDate[0]}-${localeDate[1]}-${localeDate[2]}`
+const cutTime = `${localeTime[0]}-${localeTime[1]}-${localeTime[2]}`
+
+exports.productsCatalog = products => {
```
Comment (bionanek) [inline api/helpers/pdfGenerate.js]: this is a method, so the name of it should be a verb

File: api/helpers/pdfGenerate.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const fileSystem = require("fs")
+const pdf = require("html-pdf")
+const productsCatalogTemplate = require("../helpers/pdf-templates/productsCatalogTemplate")
+
+const folderPath = "./pdfs/"
+
+const localeDate = new Date().toLocaleDateString().split("/")
+const localeTime = new Date().toLocaleTimeString().split(":")
+
+const cutDate = `${localeDate[0]}-${localeDate[1]}-${localeDate[2]}`
```
Comment (bionanek) [inline api/helpers/pdfGenerate.js]: what does "cut" stand for?

File: api/controllers/productsController.js [hunk 3]
```diff
@@ -86,3 +87,15 @@ exports.getByPriceRange = async (request, response) => {
 		response.status(error.code).json(error)
 	}
 }
+
+exports.generateProductsCatalog = async (request, response) => {
+	try {
+		const generatedPdf = await productsCatalog(request.body)
+
+		setTimeout(() => {
```
Comment (bionanek) [inline api/controllers/productsController.js:96]: why do you need timeout here?

Comment (kachmashk) [reply api/controllers/productsController.js:96]: cuz the file is being created too long so without it you'd have to click generation button twice

**Review by bionanek — APPROVED** [review]

**Pull request merged**